GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+import android.text.TextUtils;
+
+public abstract class Preconditions {
+
+    /**
+     * The goal of this class is to improve the readability and to provide some
+     * common checking method to improve the robustness of GVRF. But in some
+     * cases, this may be a significant performance cost. Try to use and further
+     * extend this class carefully with concern on performance issue.
+     * 
+     * Actually, one purpose of this class may help us avoid from using
+     * GVRJniException which is not a proper way to throw/simulate an exception
+     * with JNI crash.
+     */
+
+    /**
+     * Check that the if the parameter is {@code null}.
+     * 
+     * @param object
+     *            the parameter object can be any Java reference type, including
+     *            arrays.
+     * @param parameterName
+     *            To point out the type of data, for the use when we try to
```

Comment:
'The name of the user-supplied parameter that we are validating, so that the user can easily find the error in their code.`